test/index.spec.js

```diff
@@ -269,26 +286,30 @@ describe('multihash', () => {
 
   describe('coerceCode', () => {
     it('valid', () => {
-      /** @type {Partial<Record<import('../src/constants').HashName, import('../src/constants').HashCode>> } */
+      /** @type {Partial<Record<HashName, HashCode>> } */
       const names = {
         sha1: 0x11,
         'sha2-256': 0x12,
         'sha2-512': 0x13,
         'sha3-512': 0x14
       }
 
-      Object.keys(names).forEach((name) => {
+      // eslint-disable-next-line guard-for-in
+      for (const name in names) {
         expect(
-          mh.coerceCode(/** @type {import('../src/constants.js').HashName} */(name))
+          mh.coerceCode(/** @type {keyof typeof names} */(name))
         ).to.be.eql(
-          names[name]
+          names[/** @type {keyof typeof names} */(name)]
         )
 
         expect(
-          mh.coerceCode(names[name])
+          mh.coerceCode(/** @type {HashCode} */(names[/** @type {HashName} */(name)]))
```

Comment:
why aren't both of these things being inherited naturally from the typedef for `name` since there's a clear path to `name` and therefore `names[name]`?

Reply:
because TS is dumb and `name` in `for in` has string type. @Gozala can we improve this ?

Reply:
Something [along these lines maybe](https://www.typescriptlang.org/play?strictNullChecks=true&noUncheckedIndexedAccess=true&noUnusedLocals=true&noUnusedParameters=true&target=6&ts=4.0.5&useJavaScript=true#code/PQKhCgAIUgBAXAngBwKYBNUDNIG8BEAzgBYCGAjPpAD6RFkBMAtAwKwBs+AvpABKkkAcqQC2qKDAQoM2PAAYAHuXI1Ii8gx78SAYQD2mCcHABjPQDtC8SOdGpCkALyRQkpGjwAlVGYBO6AB5tYmExABo+AWJ9TAA+HhBgSAAKXChIDJIKAC41JXIw9IyAcizmNnZi3PUGcC4ASnBwVwk4eFQRZAAbUna8AHkAIwArH3geABVW2GRSX1E8CZ49EbHp31R4AFdfSzwAQV95xACAa1REPRwJ+KNTCytIc8QHZ2SV0ZN4eqdYlzA2tIDkdSCdnldIDcEsBkkNPvAAHTPQjvVZfeqNcBYPS+FJmSzWWxiSAQ5HJIn2DF4IqQfGPABupC6W1QThsdkIAG0KQBdGl0vRdVAIrp6ADmyUZzNQjS4QA)